contracts/modules/TransferManager/GTM/GeneralTransferManager.sol

```diff
@@ -480,41 +534,71 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
         );
     }
 
-    function _modifyKYCDataSignedMulti(
+    function _verifySignedKYCData(
         address[] memory _investor,
         uint256[] memory _canSendAfter,
         uint256[] memory _canReceiveAfter,
         uint256[] memory _expiryTime,
         uint256 _validFrom,
         uint256 _validTo,
         uint256 _nonce,
-        bytes memory _signature
+        bytes memory _signature,
+        bool _isReadOnlyCall
     )
         internal
-        returns(bool)
+        view
+        returns(bool isVerifiedSig, address signer)
     {
         if (_investor.length != _canSendAfter.length ||
             _canSendAfter.length != _canReceiveAfter.length ||
             _canReceiveAfter.length != _expiryTime.length
         ) {
-            return false;
+            return (false, address(0));
         }
 
         if (_validFrom > now || _validTo < now) {
-            return false;
+            return (false, address(0));
         }
 
         bytes32 hash = keccak256(
             abi.encodePacked(this, _investor, _canSendAfter, _canReceiveAfter, _expiryTime, _validFrom, _validTo, _nonce)
         );
 
-        if (_checkSig(hash, _signature, _nonce)) {
+        (isVerifiedSig, signer) = _checkSigView(hash, _signature, _nonce);
+        if (isVerifiedSig) {
+            if (_isReadOnlyCall) {
+                for (uint256 i = 0; i < _investor.length; i++) {
```

Comment:
Doesn't this mean that if the if statements fail in L571 or L575 the function will return (true, ...) even if `_isReadOnlyCall == true`? Not sure why we're only doing the uint64 checks when `_isReadOnlyCall == true`.